src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,239 @@
+ * "pedestrian" tag. Valid intersecting edges are edges with the same elevation(same
+ * {@link LayerTag}, {@link LocationTag}) as the area, and highway tag value not equal to foot way,
+ * pedestrian, steps and path. The pedestrian area and any valid intersecting/ overlapping edge that
+ * is not snapped to the area are flagged along with its connected edges that are within the
+ * pedestrian area.
+ *
+ * @author sayas01
+ */
+public class PedestrianAreaOverlappingEdgeCheck extends BaseCheck<Long>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList("Pedestrian area {0,number,#} is overlapping way id(s) {1}.");
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            {@link Configuration} required to construct any Check
+     */
+    public PedestrianAreaOverlappingEdgeCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    /**
+     * Checks to see whether the supplied object class type is valid for this particular check
+     *
+     * @param object
+     *            The {@link AtlasObject} you are checking
+     * @return true if it is
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Valid object for the check is a pedestrian area that has not been flagged
+        return object instanceof Area && this.isPedestrianArea(object)
+                && !isFlagged(object.getOsmIdentifier());
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Area area = (Area) object;
+        // All segments of the area
+        final List<Segment> segments = area.asPolygon().segments();
+        // Get all start and end locations of all segments in the area
+        final List<Location> locationsInSegments = Segment.asList(segments);
+        // Filter and collect all valid intersecting edges
+        final Set<Edge> filteredIntersectingEdges = Iterables
+                .stream(area.getAtlas().edgesIntersecting(area.asPolygon()))
+                .filter(edge -> this.isValidIntersectingEdge(edge, area)).collectToSet();
+        final Set<AtlasObject> overlappingEdges = new HashSet<>();
+        final Polygon areaPolygon = area.asPolygon();
+        for (final Edge edge : filteredIntersectingEdges)
+        {
+            final Location edgeStartLocation = edge.start().getLocation();
+            final Location edgeEndLocation = edge.end().getLocation();
+            // Checks if intersection is at start of the edge
+            final boolean intersectsAtStart = locationsInSegments.contains(edgeStartLocation);
+            // Checks if intersection is at end of the edge
+            final boolean intersectsAtEnd = locationsInSegments.contains(edgeEndLocation);
+            // Filter the segments with the intersecting points
+            final Set<Segment> filteredSegments = segments.stream()
```

Comment:
Is there a reason we want to pre-filter like this, as opposed to just feeding all the segments to this.isNotSnappedToSegments?

Reply:
I think the computation for filtering and then checking for isNotSnappedToSegments on filtered edges will be the same as isNotSnappedToSegments on un filtered segments. So removing the filter. Nice suggestion!